File: bailoteca-backend/src/main/java/com/bailoteca/dto/RegistroRequest.java
package com.bailoteca.dto;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import lombok.Data;

/**
 * Clase que representa una solicitud de registro de usuario.
 * Contiene los campos necesarios para crear un nuevo usuario: nombre, apellido, correo y contraseña.
 * 
 * @author Tony Aragón
 * @version 1.0
 * @since 1.0
 */
@Data
public class RegistroRequest {

    @NotBlank
    private String nombre;

    @NotBlank
    private String apellido;

    @Email
    @NotBlank
    private String correo;

    @NotBlank
    private String contrasenna;
}

File: bailoteca-frontend/app/src/main/java/com/example/bailotecaapp/model/dto/RegistroRequest.kt
package com.example.bailotecaapp.model.dto

import com.example.bailotecaapp.model.enums.Rol
import kotlinx.serialization.Serializable

/**
 * DTO para enviar datos de registro al backend.
 */
@Serializable
data class RegistroRequest(
    val nombre: String,
    val apellido: String,
    val correo: String,
    val contrasenna: String
)